### PR TITLE
encryption: add key controller

### PR DESCRIPTION
**Pull request by sttts**

### CODE REVIEW

**Comment by sttts** [comment]
/hold

updating godoc

File: pkg/operator/encryption/key_controller.go [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+	"github.com/openshift/library-go/pkg/operator/events"
+	operatorv1helpers "github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+const encWorkKey = "key"
+
+// encryptionKeyController watches secrets in openshift-config-managed
+// to determine if a new AES-256 bit encryption key should be created.
+// It finds the secrets that contain the keys using the encryptionSecretSelector.
+// There are distinct keys for each resource that is encrypted per encryptedGRs.
```
Comment (p0lyn0mial) [inline pkg/operator/encryption/key_controller.go]: update the comment, there is only one write key for all `encryptedGRs`

File: pkg/operator/encryption/key_controller.go [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+//   - secrets in openshift-config-managed
+//   - pods in target namespace
+//   - secrets in target namespace
+// * computes a new, desired encryption config from encryption-config-<revision>
+//   and the existing keys in openshift-config-managed.
+// * derives from the desired encryption config whether a new key is needed due to
+//   - encryption is being enabled via the API or
+//   - a new to-be-encrypted resource shows up or
+//   - the EncryptionType in the API does not match with the newest existing key or
+//   - based on time (once a week is the proposed rotation interval).
```
Comment (p0lyn0mial) [inline pkg/operator/encryption/key_controller.go]: and due to a new external reason - https://github.com/openshift/cluster-kube-apiserver-operator/pull/584/commits/0a5781a810336d29783edb1580e4783e61bf1cbb#diff-b44a5c1483fa3b6714560797ab0f1b4bR288

File: pkg/operator/encryption/key_controller.go (hunk 1 above)
Comment (sttts) [reply pkg/operator/encryption/key_controller.go]: done

File: pkg/operator/encryption/key_controller.go (hunk 2 above)
Comment (sttts) [reply pkg/operator/encryption/key_controller.go]: fixed

**Pull request merged**